lib/pluginmanager/unpack.rb

```diff
@@ -0,0 +1,47 @@
+class LogStash::PluginManager::UnPack < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "unpack a packaged tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "unpack a packaged  zip file", :default => LogStash::Environment.windows?
+  option "--force", :flag, "Override any previously generated cache", :default => false
+
+  parameter "file", "the package file name", :attribute_name => :package_file, :required => true
+
+  def execute
+    puts("Unpacking #{package_file} to be used as a plugins local cache")
+    if force?
+      require "fileutils"
+      FileUtils.rm_rf(LogStash::Environment::CACHE_PATH)
+    end
```

Comment:
This look like duplicated code from https://github.com/elastic/logstash/pull/3404/files#diff-ccfd432c822c61f42c736fcd8bf0e000R30 Maybe we could move the condition in the #validate_cache_location?